docs/BindingWithGrammar.md

```diff
@@ -0,0 +1,114 @@
+# Binding with Grammar
+
+[vscode-xml](https://github.com/redhat-developer/vscode-xml) provides support for automatic XML schema/grammar binding through generation and/or using an existing `.xsd/.dtd` file.
+
+[Validation with XSD grammar](Validation.md#validation-with-xsd-grammar) and [Validation with DTD grammar](Validation.md#validation-with-dtd-grammar) provide more information on manual binding of an XML document.
+
+In the following sections, we would like to bind the XML document `foo.xml`:
+
+```xml
+<foo>
+  <bar />
+</foo>
+```
+
+with an associated grammar file.
+
+## Binding with Existing Grammar
+
+Consider an XSD document, `foo.xsd` file (in the same directory as foo.xml) defined as follows :
+
+```xsd
+<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:element name="foo">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="bar" />
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+</xs:schema>
+```
+
+This file (or a corresponding `.dtd` file) can be bound using a [binding wizard](#the-xml-binding-wizard), which can be triggered by [Command](#command), [CodeLens](#codelens) or [Quick Fix/CodeAction](#quick-fix/codeaction).
+
+### The XML Binding Wizard
+
+After opening the binding wizard using one of the methods mentioned above, a dropdown with the following 2 options will appear at the Command Palette:
+
+![Binding Wizard Dropdown](./images/BindingWithGrammar/BindingWizardDropdown.png)
+
+1. Standard (xsi, DOCTYPE)
+
+This option will bind the grammar file by adding the `xmlns:xsi`(with the value as "http://www.w3.org/2001/XMLSchema-instance") and `xsi:noNamespaceSchemaLocation`(with the value as the path to the grammar file) attributes to the root tag:
+
+```xml
+<foo xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:noNamespaceSchemaLocation="foo.xsd">
+    <bar />
+</foo>
+```
+
+2. XML Model association
+
+This option will bind the grammar file by adding the `xml-model` tag at the beginning of the file, with an `href` value as the path to the grammar file:
+
+```xml
+<?xml-model href="foo.xsd"?>
+<foo>
+    <bar />
+</foo>
+```
+
+After selecting an option, you can select the desired grammar file using your file explorer.
+
+### Command
+
+The command "XML: Bind to grammar/schema file" can be executed from the VSCode command palette, as well as be bound to a shortcut.
+
+![Bind To Grammar Command](./images/BindingWithGrammar/BindToGrammarCommand.png)
+
+If the current XML document is not bound using an `xml-model` tag or a root element doesn't have a `xsi:noNameSpaceLocation` attribute, the command will have the user select a existing `.xsd/.dtd` grammar file to bind to the XML document.
+
+Otherwise, an error will be thrown on the client if the XML document already has a bound grammar/schema.
+
+### CodeLens
+
+When CodeLens is enabled, an unbound XML document will display a CodeLens above the root element as follows:
```

Comment:
... When CodeLenses are enabled ...

I think it might be worthwhile to link to the setting to enable code lens in `Preferences.md#code-lens`